### PR TITLE
Fixes for edge cases - Swagger JSON and YAML

### PR DESCRIPTION
**Pull request by JackuB**

### CODE REVIEW

**Comment by honzajavorek** [comment]
:+1:

**Comment by JackuB** [comment]
Seems like I don't have rights for deploy - could you pls @honzajavorek ?

File: src/deckardcain.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const REFRACT_PARSE_RESULT_ELEMENT_YAML = /[\uFEFF]?\n?\s*element: ?["']parseRes
  * @returns {string|null} Media type of given file.
  */
 function identify(source) {
-  if (source.match(API_BLUEPRINT_HEADER)) {
+  if (source.substring(0, 11).match(API_BLUEPRINT_HEADER)) {
```
Comment (freaz) [inline src/deckardcain.js]: Why 11?

Just simple two lines brakes it ```



FORMAT: 1A
HOST: http://polls.apiblueprint.org/

# Some example

Polls is a simple API allowing consumers to view polls and vote in them.

## Questions Collection [/questions]

```
or

```
[comment]: I broke it!

FORMAT: 1A
HOST: http://polls.apiblueprint.org/

# Some example

Polls is a simple API allowing consumers to view polls and vote in them.

## Questions Collection [/questions]

### List All Questions [GET]
```

Comment (JackuB) [inline src/deckardcain.js]: Metadata section should start at the beginning of the document - https://github.com/apiaryio/api-blueprint/blob/master/API%20Blueprint%20Specification.md#definition-3

Comment (freaz) [inline src/deckardcain.js]: Ok then I trust you. _(but not to users)_

Comment (honzajavorek) [inline src/deckardcain.js]: +1 to being super suspicious

Comment (JackuB) [inline src/deckardcain.js]: But regex we are using everywhere IS checking beginning of the string. And afaik we have no idea what would actually happen to our tools if you'd pass invalid, according to spec, APIB.

Only false negative scenario thing that _could_ happen is FORMAT not being first. But all our docs and examples have it at the beginning of document.

**Pull request merged**